src/chain-libs/chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -931,14 +929,12 @@ impl Ledger {
                     return Err(Error::StakeDelegationSignatureFailed);
                 }
 
-                let (new_ledger_, _fee) =
-                    new_ledger.apply_transaction(&fragment_id, &tx, block_date)?;
```

Comment:
are there any potential adversarial implications of removing `BlockDate` . For example, it seems to be used to verify if the TX date is valid.

Reply:
Expired date transaction's field is mostly used as an only way how to notify the user that a particular transaction will not submitted anymore into the blockchain. So this feature partially replaces transaction's error propagation which is missed.
Darius comment:
"
this is due the fact that we don't have error propagation through the network. Otherwise invalid fragments could congest the network. In our network, a follower node is accepting transaction and since it cannot mint a block just propagates it to the leader. If transaction is put in the block then follower node will receive update but as a block not transaction (once it receive new block , it unwraps all transactions from it and checks if any transaction currently residing in its mempool wasn't minted in the block. If yes then it updates mempool). However, there is a problem when leader rejects fragment. Leader node is silently rejecting fragments and not sending it back to follower, also failed transaction are not minted in the block. Therefore there was an idea of expiry time, after which follower node knows that transaction for sure will be rejected (even if it's correct but unfortunate enough to be pending all the time). It helps voting app to show user that transaction is failed, so it need to be resend, and since on each resend operation voting app sync with network it can fix a problem with wrong spending counter etc.
"

As it is discussed before and aligned that we want to remove this inconvenient for user field, and I assume later we are going to replace it with the proper error propagation mechanism.

Reply:
ok, maybe we should create a ticket in relation to adding error propagation in this case.